### DISCUSSION TITLE
Discussion about (possible) "traffic_sign" quest

### DISCUSSION THREAD

**Discussion by mcliquid** [streetcomplete/StreetComplete#3805] [Ideas]
Since I recently started to enter the traffic signs for paths and streets using Mapillary photos, it occurred to me to be able to map this via StreetComplete as well.

Now I would have assumed that there is already an issue for this, but could not find anything at least using the searchterm "[traffic_sign](https://taginfo.openstreetmap.org/keys/traffic_sign)".

Therefore, here is the question for discussion: Have there already been considerations to ask for traffic signs for individual highway=* ways?

I am aware that there will be no signs for most of the ways or none that directly match the way. This would then require a traffic_sign=no to be entered.
_Interestingly, the use of [traffic_sign=none](https://taginfo.openstreetmap.org/tags/traffic_sign=none) is significantly higher at 5207 than [traffic_sign=no](https://taginfo.openstreetmap.org/tags/traffic_sign=no) which has just 314 uses._

Also, of course, the choice of selectable traffic signs is almost infinite, which would make such a quest very complex for users.
For [traffic_sign=DE:*](https://taginfo.openstreetmap.org/keys/traffic_sign#values) alone there are currently over 7500 possible values through sign combinations.

It might be possible to query the exact signage only for very specific tagging combinations. An example could be to query for all highway=cycleway and bicycle=* routes in Germany whether it is DE:240, DE241, DE237, DE239+1022-10 or none of them.
According to the entered signage, the tagging for the way would then of course also change.

So far my thoughts about it. I would be very interested in what other StreetComplete contributors think about this. Many thanks!

**Comment by matkoniecz** [comment]
>  Have there already been considerations to ask for traffic signs for individual highway=* ways?

In which cases it would be a valid tagging? Is it documented somewhere on OSM Wiki?

**Reply by mcliquid** [reply]
> In which cases it would be a valid tagging?

To be honest, even if it is German, [this page](https://wiki.openstreetmap.org/wiki/DE:Verkehrszeichen_in_Deutschland) should be enough for that. But there would be also a page for [UK](https://wiki.openstreetmap.org/wiki/Road_signs_in_the_United_Kingdom) 

There are nearly one million uses worldwide - even in [your hometown](https://overpass-turbo.eu/s/1goT).

For more examples have a look [here](https://wiki.openstreetmap.org/wiki/Key:traffic_sign#Tagging
) or [here](https://wiki.openstreetmap.org/wiki/DE:Key:traffic_sign#Werte)

To see the tagging used in OSM, check out [Munich](https://overpass-turbo.eu/s/1goR) or [Amsterdam](https://overpass-turbo.eu/s/1goS).

> Is it documented somewhere on OSM Wiki?

Poland Wiki: https://wiki.openstreetmap.org/wiki/Pl:Road_signs_in_Poland
English Wiki: https://wiki.openstreetmap.org/wiki/Key:traffic_sign
German Wiki: https://wiki.openstreetmap.org/wiki/DE:Key:traffic_sign
German traffic_sign Page: https://wiki.openstreetmap.org/wiki/DE:Verkehrszeichen_in_Deutschland
Tagging Tool (for Germany): https://osmtools.de/traffic_signs/
A list with all countries with JOSM presets: https://wiki.openstreetmap.org/wiki/Key:traffic_sign#Presets.2C_styles.2C_libraries.2C_statistics_and_maps_with_traffic_signs_with_IDs_per_country
A list with all countries with traffic_sign usage and wiki link: https://wiki.openstreetmap.org/wiki/DE:Key:traffic_sign#Liste_der_Kennungen_.28IDs.29_nach_Staat

**Reply by matkoniecz** [reply]
> To be honest, even if it is German, [this page](https://wiki.openstreetmap.org/wiki/DE:Verkehrszeichen_in_Deutschland) should be enough for that. But there would be also a page for [UK](https://wiki.openstreetmap.org/wiki/Road_signs_in_the_United_Kingdom)

In which cases traffic signs would be tagged on `highway=*` ways? Note that there are already `maxheight` `maxspeed` `maxheight` `oneway` quests - asked in places where it makes sense, not on every single `highway=*`

In which case some new quest would make sense?

> There are nearly one million uses worldwide - even in [your hometown](https://overpass-turbo.eu/s/1goT).

Thanks, I opened https://www.openstreetmap.org/note/3062955 https://www.openstreetmap.org/note/3062957 as German codes of traffic signs in Poland are definitely incorrect.

**Reply by mcliquid** [reply]
Thanks for your response!

>  In which cases traffic signs would be tagged on highway=* ways?

A possible quest would be: "Is this cycleway signed?" on all highway=cycleway and all highway=path with bicycle=yes|designated. Then a list of all possible traffic signs for the specific county is shown. 
In Germany this would be DE:240, DE241, DE237, DE239+1022-10. Depending on the users selection the traffic_sign tag is added and the tagging at all could be changed.

**Reply by matkoniecz** [reply]
Are you sure that in Germany it is recommended and desired to tag it on every single cycleway? It is definitely not a standard practice in Poland.

**Reply by mcliquid** [reply]
Most cycleways should have official signage. Otherwise, it's the same as the quest "Is this path lit?" or "What is the speed limit here?". There will not always be signage. 
However, if there is, the quest is an asset to OSM. If there is no signage, then this is marked as with lit=no.

If StreetComplete only accepts quests that must always include valid tagging, I'm happy to withdraw this discussion of a new quest.

Maybe some more users are open to start a dialog here :)

**Reply by matkoniecz** [reply]
https://github.com/streetcomplete/StreetComplete/wiki/Adding-new-Quests-to-StreetComplete#1-decide-whether-the-idea-works-out-with-streetcomplete

> 🚧 Established tags only: No new or unestablished tags should be introduced through StreetComplete. Establishing tags must remain a community process and not be dictated by software implementation.

> 🤔 No unanswerable quests: All generated quests need to be actually answerable (no false-positives). This means that any answer given by the user must result in something being tagged. For example, a quest that asks for the website of a place must be able to tag the element somehow if the user answers that the place has no website - otherwise, the next user will be asked the same question. Sometimes, due to the nature of how things are tagged in OSM (such as the one given in the example) it is unfortunately simply not possible to fulfill this.

**Reply by wielandb** [reply]
Hey!
I apprechiate the idea, because I am something of a traffic_sign-Nerd, you could say. :D
I was also searching for a way to add `traffic_sign`s on `highway=*`s while on the go, so I developed my own editor! 
While the editor itself is not open source (yet), the definitions my app uses already are!
You can see the signs (and their tagging) here: https://github.com/wielandb/osm-beschilderung/blob/main/signs.md

> Are you sure that in Germany it is recommended and desired to tag it on every single cycleway? It is definitely not a standard practice in Poland.

I'm not sure I understand what you mean by "desired/recommended". The key `traffic_sign` just says "It is a fact that the traffic sign XYZ is located at this way (and applys to it)". It is not tagging that is "implied" in any way. The tagging (or lack thereof) represents what is physically present. I don't want to use the the word "recommended", but if the german traffic sign XYZ is pysically present, `traffic_sign:DE=XYZ` is definitely correct tagging. (Edit: I think it could be viewed similar to `maxspeed:type`. It tells you where the tagging comes from)

There _are_ use cases, but even I have to admit they are minimal... 
(But to name some: They can be a great help for veryfing access-restrictions; knowing which paths are "officially labled footways" by the government can be helpful; I personally like recording where "wrong" traffic sign combinations are placed)

After saying all that, and after outing myself as a vivid `traffic_sign`-lover I have to say that even I am **strongly against** using StreetComplete to tag traffic signs. If one looks at the heterogenity of signs that can apply to a way, it would be hard to design a filter to capture `highway`s that are likely enough to have a traffic sign that the amout of `traffic_sign=no` would be kept to an acceptable minimum.
The next problems would be combinations of signs. The traffic_sign tool that @mcliquid linked has a great algorithm for determining how the tags for a way change if you add extra signs, it would need to be reimplemented in SC and it would be too much work for very little gain.
(Even I was too lazy to reimplement the algorithm in an app _specifically made for the purpose of tagging traffic signs_ 😁, so implementing it in SC would be complete overkill)

To bring another example, I made some drafts how traffic_signs could be used in StreetComplete, but I soon realized that that would also be too much work considering you'd have to add every country's version of theese signs.

![sidewalk](https://user-images.githubusercontent.com/29458682/155619916-f69ba189-1f90-459c-b04e-f46193ac2118.png)
 
So, while my heart is absolutely with you @mcliquid and I would love to see utilization of traffic_sign in SC, it's probably not a good idea. 😔
I just hope I could convince @matkoniecz that traffic_sign tagging is not useless. 😳

**Comment by mnalis** [comment]
I do not think that would be a good idea generally. There are, as you say, a lot of possible traffic signs, so it would be a nightmare for user to find correct one (even if they all looked the same in all countries, which they do not, so this would also be a nightmare for developers to implement in worldwide fashion). 

And, there might be multiple signs. And, they might relate to different lengths of the road (for example `road works` might be for next 100m, while `no overtaking` might be until next crossing, etc.) 

Also,  StreetComplete does not currently allow to create tags on-demand (like [Vespucci](https://wiki.openstreetmap.org/wiki/Vespucci) for example does), but only in predefined quests. So, it seems it would need to need to be asked on all roads, which is way too spammy. 

Note that SC **_does_** have quests which record the fact whether the value was **_posted by sign_** or not (for example, `maxspeed` quest you mention, will specifically ask if the limit is from the explicit `speed restriction` sign or from some other source, like nonmarked urban speed limit). However, even that "simple" quest ("simple" as it is only about one or two signs, and not hundreds of them) is so complex for the user, that  the quest had to be disabled by default.

So I think StreetComplete is ill-suited solution for this, it would be better:
- long press on the screen on location of sign, and leave a picture of sign and explanation of direction (as that information is not yet recorded in pictures taken by SC, and it is important to know for which direction sign is - see https://github.com/streetcomplete/StreetComplete/issues/2880)
- long press on screen on location of sign, and open location in some other full-fledged editor (like [Vespucci](https://wiki.openstreetmap.org/wiki/Vespucci)), where you can than record that information about sign directly.

**Reply by matkoniecz** [reply]
> long press on the screen on location of sign, and leave a picture of sign and explanation of direction

though as StreetComplete is not showing street signs you could be creating notes about what is tagged

also, notes about things of very minor importance are a bit dubious, I would do this only of I would process the notes myself

> long press on screen on location of sign, and open location in some other full-fledged editor (like [Vespucci](https://wiki.openstreetmap.org/wiki/Vespucci))

+1
